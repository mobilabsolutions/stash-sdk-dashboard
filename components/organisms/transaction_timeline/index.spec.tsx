import { testRender } from '../../../test_utils'
import TransactionTimeline from './index'

it('CaptureForm Should Render', () =>
  testRender(TransactionTimeline, {
    timeline: [
      {
        action: 'REFUND',
        amount: 3470,
        createdDate: '2019-06-26 14:49:18',
        reason: 'No se',
        status: 'SUCCESS'
      },
      {
        action: 'AUTH',
        amount: 3470,
        createdDate: '2019-06-21 17:18:16',
        reason: 'Test Payment',
        status: 'SUCCESS'
      }
    ]
  }))

it('CaptureForm Should Render preauth explaination', () =>
  testRender(TransactionTimeline, {
    timeline: [
      {
        action: 'PREAUTH',
        amount: 3470,
        createdDate: '2019-06-26 14:49:18',
        reason: 'No se',
        status: 'SUCCESS'
      },
      {
        action: 'AUTH',
        amount: 3470,
        createdDate: '2019-06-21 17:18:16',
        reason: 'Test Payment',
        status: 'FAIL'
      }
    ]
  }))
