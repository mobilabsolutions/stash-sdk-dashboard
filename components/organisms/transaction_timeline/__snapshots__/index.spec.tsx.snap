// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionTimeline Should Render 1`] = `
<div>
  <h2
    className="h2-m2oto9-0 detail_view__Title-inrqp5-1 ibbisR"
  >
    Timeline
  </h2>
  <div
    className="detail_view__ContentWrapper-inrqp5-0 iyMXGu"
    style={Object {}}
  >
    <table>
      <tbody>
        <tr>
          <td
            className="transaction_timeline__TimeActionItem-sc-1qjudet-0 bxHiTk"
          >
            <span>
              6/26/2019, 2:49:18 PM
              <span
                className="item-details"
              >
                 - 
                Refund
                 
                Success
                 - 
                No se - 
                <span>
                  € 34.70
                </span>
              </span>
            </span>
          </td>
        </tr>
        <tr>
          <td
            className="transaction_timeline__TimeActionItem-sc-1qjudet-0 bxHiTk"
          >
            <span>
              6/21/2019, 5:18:16 PM
              <span
                className="item-details"
              >
                 - 
                Authorisation
                 
                Success
                 - 
                Test Payment - 
                <span>
                  € 34.70
                </span>
              </span>
            </span>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`TransactionTimeline Should ommit empty reasons 1`] = `
<div>
  <h2
    className="h2-m2oto9-0 detail_view__Title-inrqp5-1 ibbisR"
  >
    Timeline
  </h2>
  <div
    className="detail_view__ContentWrapper-inrqp5-0 iyMXGu"
    style={Object {}}
  >
    <table>
      <tbody>
        <tr>
          <td
            className="transaction_timeline__TimeActionItem-sc-1qjudet-0 bxHiTk"
          >
            <span>
              6/26/2019, 2:49:18 PM
              <span
                className="item-details"
              >
                 - 
                Refund
                 
                Success
                 - 
                
                <span>
                  € 34.70
                </span>
              </span>
            </span>
          </td>
        </tr>
        <tr>
          <td
            className="transaction_timeline__TimeActionItem-sc-1qjudet-0 bxHiTk"
          >
            <span>
              6/21/2019, 5:18:16 PM
              <span
                className="item-details"
              >
                 - 
                Authorisation
                 
                Success
                 - 
                Test Payment - 
                <span>
                  € 34.70
                </span>
              </span>
            </span>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
